source/simulator/core.cc

```diff
@@ -150,99 +150,102 @@ namespace aspect
                     std_cxx14::make_unique<NewtonHandler<dim>>() :
                     nullptr),
     post_signal_creation(
-      std::bind (&internals::SimulatorSignals::call_connector_functions<dim>,
-                 std::ref(signals))),
-    volume_of_fluid_handler (parameters.volume_of_fluid_tracking_enabled ?
-                             std_cxx14::make_unique<VolumeOfFluidHandler<dim>> (*this, prm) :
-                             nullptr),
-    introspection (construct_variables<dim>(parameters, signals, melt_handler), parameters),
-    mpi_communicator (Utilities::MPI::duplicate_communicator (mpi_communicator_)),
-    iostream_tee_device(std::cout, log_file_stream),
-    iostream_tee_stream(iostream_tee_device),
-    pcout (iostream_tee_stream,
-           (Utilities::MPI::
-            this_mpi_process(mpi_communicator)
-            == 0)),
-
-    statistics_last_write_size (0),
-    statistics_last_hash (0),
-
-    computing_timer (mpi_communicator,
-                     pcout,
-                     TimerOutput::never,
-                     TimerOutput::wall_times),
-    initial_topography_model(InitialTopographyModel::create_initial_topography_model<dim>(prm)),
-    geometry_model (GeometryModel::create_geometry_model<dim>(prm)),
-    // make sure the parameters object gets a chance to
-    // parse those parameters that depend on symbolic names
-    // for boundary components
-    post_geometry_model_creation_action (std::bind (&Parameters<dim>::parse_geometry_dependent_parameters,
-                                                    std::ref(parameters),
-                                                    std::ref(prm),
-                                                    std::cref(*geometry_model))),
-    material_model (MaterialModel::create_material_model<dim>(prm)),
-    gravity_model (GravityModel::create_gravity_model<dim>(prm)),
-    prescribed_stokes_solution (PrescribedStokesSolution::create_prescribed_stokes_solution<dim>(prm)),
-    adiabatic_conditions (AdiabaticConditions::create_adiabatic_conditions<dim>(prm)),
+      [&]()
+  {
```

Comment:
astyle sadly gets confused with the indentation here. Should we leave the bind() call in here?

Reply:
Is there a real downside of using bind? If not I would personally prefer the readability for now. Also in the other places where bind has been replaced.